### PR TITLE
sct_get_centerline: use the new viewer

### PR DESCRIPTION
**Pull request by charleygros**
`sct_get_centerline` currently uses `sct_viewer.ClickViewerPropseg()` instead of the new `spinalcordtoolbox.gui.*` viewer.

This Pull Request aims at integrating the new viewer into `sct_get_centerline`.

To test it, please run:
```
cd example_data/t2
sct_get_centerline -i t2.nii.gz -method viewer
sct_image -i t2.nii.gz -setorient SAL -o t2SAL.nii.gz
sct_get_centerline -i t2SAL.nii.gz -method viewer
```

This code was in `sct_deepseg_sc` --> I moved it in `sct_get_centerline` --> that s why both `sct_deepseg_sc` and `sct_deepseg_lesion` now called functions from `sct_get_centerline`.

To test it, please run:
```
cd example_data/t2
sct_deepseg_sc -i t2.nii.gz -c t2 -centerline viewer
```

This PR fixes the issue #1837.

### CODE REVIEW

**Comment by jcohenadad** [comment]
i still find issues if the number of points is too low (switches to hanning, but we don't know because verbose is set to False). For this PR, i would suggest to only introduce the new viewer, without perturbating too much the behaviour (which requires extensive testing on larger set of images). So i would suggest the following modifs:
- set verbose to "verbose" (instead of 0) when calling smooth_centerline
- set nurbs_pts_number=3000 (as was the case before)
- use the same number of points & gap as before (was calculated based on max number of slices)
- if not too much trouble, replace `_from_viewerLabels_to_centerline` by the already-existing `sct_process_segmentation.extract_centerline` (which does the same thing)

happy to help :-)

**Comment by charleygros** [comment]
@jcohenadad: Thank you for your help.
The last commits do:
- replace `_from_viewerLabels_to_centerline` by the already-existing `sct_process_segmentation.extract_centerline`
- set verbose to "verbose" (instead of 0) when calling `smooth_centerline` in `sct_process_segmentation.extract_centerline`
- use the same number of points & gap as before (calculated based on max number of slices)

I tested it on the example_data with the default param (gap=10) and it seems to work well.
Please let me know your thoughts

File: scripts/sct_get_centerline.py [hunk 1]
```diff
@@ -12,39 +12,39 @@
...
 import spinalcordtoolbox.image as msct_image
 from spinalcordtoolbox.image import Image
-from msct_types import Centerline
-from sct_viewer import ClickViewerPropseg
-from sct_straighten_spinalcord import smooth_centerline
+from sct_process_segmentation import extract_centerline
 
 
-def viewer_centerline(image_input_reoriented, interslice_gap, verbose):
+def _call_viewer_centerline(fname_in, interslice_gap=10):
```
Comment (jcohenadad) [inline scripts/sct_get_centerline.py]: i find it a bit too dense now, how about setting a compromise with `interslice_gap=20`?

Comment (charleygros) [reply scripts/sct_get_centerline.py]: I agree!

**Comment by charleygros** [comment]
@jcohenadad: By increasing the default gap to 20.0mm, I observed 'not smooth centerline' again.
In the last commits, I proposed to increase the number of points used by the nurbs (3000 to 8000) --> looks much better. However it takes more time.

Another approach would be to leave gap=20 and `nurbs_pts_number=3000` but to send a message to the user if: https://github.com/neuropoly/spinalcordtoolbox/blob/master/scripts/sct_straighten_spinalcord.py#L160
--> asking the user to decrease the gap used by the viewer.

**Comment by jcohenadad** [comment]
@charleygros let's go with your latest commit. We definitely don't want to have an approach where a message is sent, asking the user to make a decision (99.99% of our users don't read the terminal output).

**Pull request merged**